pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ProducerConfigurationData.java

```diff
@@ -189,6 +189,8 @@ public class ProducerConfigurationData implements Serializable, Cloneable {
     )
     private CompressionType compressionType = CompressionType.NONE;
 
+    private int compressMinMsgBodySize = 4 * 1024; // 4kb
```

Comment:
What's the rationale for the default value?

Reply:
- It is default value for rocket mq.
https://github.com/apache/rocketmq/blob/dd62ed0f3b16919adec5d5eece21a1050dc9c5a0/client/src/main/java/org/apache/rocketmq/client/producer/DefaultMQProducer.java#L117
- And in my testing (against Pulsar Producer), when the message body is in the 3k-5k range, the CPU usage of compressing the same data starts to drop significantly and the compression rate increases significantly.

Reply:
This data may be more convincing.
https://docs.google.com/spreadsheets/d/1pFOAgxVYsos38oraeva1_RHC9P4J3oN2Oj2fQ65L8OI/edit?gid=0#gid=0

Reply:
Thanks for sharing the reasoning.